lib/Nusoap/class.wsdl.php

```diff
@@ -755,8 +755,8 @@ function getTypeDef($type, $ns) {
     * @access private
     */
     function webDescription(){
-		if (isset($_SERVER)) {
-			$PHP_SELF = $_SERVER['PHP_SELF'];
+		if ( isset( $_SERVER['PHP_SELF'] ) ) {
```

Comment:
You may want to revert back to `if ( isset( $_SERVER ) ) {` or use a ternary operation to set `$PHP_SELF` to something regardless; there is no check when used later.

Reply:
Added ternary to set PHP_SELF and refactored conditional check for setting error if PHP_SELF is empty.